scrapy/downloadermiddlewares/stats.py

```diff
@@ -1,7 +1,32 @@
 from scrapy.exceptions import NotConfigured
+from scrapy.utils.python import global_object_name, to_bytes
 from scrapy.utils.request import request_httprepr
-from scrapy.utils.response import response_httprepr
-from scrapy.utils.python import global_object_name
+
+from twisted.web import http
+
+
+def get_header_size(headers):
+    if headers is None:
+        return None
+    size = 0
+    for key, value in headers.items():
+        if isinstance(value, bytes):
+            size += len(b": ") + len(key) + len(value)
+        elif isinstance(value, (list, tuple)):
+            for v in value:
+                size += len(b": ") + len(key) + len(v)
+    return size + len(b'\r\n') * (len(headers.keys()) - 1)
+
+
+def get_status_size(response_status):
+    values = [
+        9,
+        len(to_bytes(str(response_status))),
+        1,
+        len(to_bytes(http.RESPONSES.get(response_status, b''))),
+        2,
+    ]
+    return sum(values)
```

Comment:
```suggestion
    return (
        9  # b"HTTP/1.1 "
        + len(to_bytes(str(response_status)))
        + 1  # " "
        + len(to_bytes(http.RESPONSES.get(response_status, b"")))
        + 2  # b"\r\n"
    )
```
Adding comments for future reference. Also, since we're optimizing memory usage, building a list (and assigning it to a variable) has an unnecessary overhead (even more so than a tuple, because of their mutability).
Going even further, the integers could be consolidated, but that might be counter-productive in terms of readability.

Reply:
@elacuesta related codelines changed in https://github.com/GeorgeA92/scrapy/commit/d7b3cbfbaab7c040d306bf9be0a64c2e144accc2